### ISSUE TITLE
[INSTALL] Install hpc-stack with Intel 2021.3.0 and impi 2021.3.0

### ISSUE THREAD

**Issue by junwang-noaa** [NOAA-EMC/hpc-stack#378]
Note: Please search to see if an issue already exists for the software you are requesting to be installed.

**Which software in the stack would you like installed?**
The hpc-stack used by ufs-weather-model.

**What is the version/tag of the software?**
hpc/1.1.0
with Intel 2021.3.0 and impi 2021.3.0, following libraries are required:

jasper/2.0.22
zlib/1.2.11
png/1.6.35

hdf5/1.10.6
netcdf/4.7.4
pio/2.5.2
esmf/8_2_1bs04
fms/2021.03

bacio/2.4.1
crtm/2.3.0
g2/3.4.2
g2tmpl/1.10.0
ip/3.3.3
sp/2.3.3
w3nco/2.4.1

gftl-shared/v1.3.3
yafyaml/v0.5.1
mapl/2.11.0-esmf-8_2_0

Please provide the version/tag of the software to be installed.

**What compilation options would you like set?**
fms/2021.03 needs to build with avx on hera (see fms/2021.03-avx under hpc/1.1.0 in /scratch2/NCEPDEV/nwprod/hpc-stack/libs/hpc-stack/modulefiles/stack

**Which machines would you like to have the software installed?**
On hera first, we will test it and confirm it is working. Then it will be installed on jet, gaea, orion, dell, cray and wcoss2

**Any other relevant information that we should know to correctly install the software??**
Please describe in as much detail as possible.

**Additional context**
Add any other context or screenshots about the install request here.

**Comment by kgerheiser** [comment]
FYI I don't see an `intel/2021.3` on Orion, just `intel/2021.2`.

**Comment by Hang-Lei-NOAA** [comment]
This is starting on Hera. I will wait for the quota limit is being
increased, and then install it on hera.

**Comment by junwang-noaa** [comment]
Thanks. Intel 2021.2 is fine, we need to move to Intel 2021.x version.

**Comment by junwang-noaa** [comment]
@jkbk2004 FYI.

**Comment by jkbk2004** [comment]
Here is the test result of intel2021.2 with the control/butterfly case of -DAPP=ATM. Package options in the intel22021.2 run can be stayed only for the updates on the compiler and cmake. Will continue to follow up. 
[https://docs.google.com/presentation/d/1h91HqW80W3onXFS0brYIh-K7Dl42h4HK7R3fJIvK2CA/edit#slide=id.p](url)

**Comment by jkbk2004** [comment]
[control-intel21.pdf](https://github.com/NOAA-EMC/hpc-stack/files/8044945/control-intel21.pdf)

**Comment by jkbk2004** [comment]
Here is the test result of intel2021.2 with the control/butterfly case of -DAPP=ATM. Package options in the intel22021.2 run can be stayed only for the updates on the compiler and cmake. Will continue to follow up.
[control-intel21.pptx](https://github.com/NOAA-EMC/hpc-stack/files/8044951/control-intel21.pptx)

**Comment by jkbk2004** [comment]
Here is the test result of intel2021.2 with the control/butterfly case of -DAPP=ATM. Package options in the intel22021.2 run can be stayed only for the updates on the compiler and cmake. Will continue to follow up.
[control-intel21.pptx](https://github.com/NOAA-EMC/hpc-stack/files/8045265/control-intel21.pptx)

**Comment by junwang-noaa** [comment]
@Hang-Lei-NOAA May I ask if the disk issue is resolved? When hpc-stack with intel/impi 2021 can be installed?

**Comment by kgerheiser** [comment]
@Hang-Lei-NOAA is on leave this week

**Comment by kgerheiser** [comment]
@junwang-noaa it was recommended by an RDHPCS admin that we should use Intel 2022. 2021 contains the log4j vulnerability.

**Comment by junwang-noaa** [comment]
Sure. If you have the library installed with Intel 2022, we can do some testing with ufs-weather-model. Currently we have one platform (cheyenne) running with Intel 2021.02 and it works fine, we haven't tried Intel 2022 yet.

**Comment by Hang-Lei-NOAA** [comment]
Hi, Jun
I will install the library installed with Intel 2022 on hera this afternoon
and evening, and you can try it tomorrow.
Thanks,
Hang

**Comment by jkbk2004** [comment]
@kgerheiser @junwang-noaa @Hang-Lei-NOAA let me know if intel2022 option is available on orion
. I can quickly launch out RT test.

**Comment by kgerheiser** [comment]
@Hang-Lei-NOAA just FYI, Hera will be undergoing maintenance tomorrow.

**Comment by kgerheiser** [comment]
The 2022 compilers aren't available on Orion, but I put in a request to have them installed.

**Comment by kgerheiser** [comment]
Orion admins are in the process of installing the new compilers.

**Comment by junwang-noaa** [comment]
@Hang-Lei-NOAA May I ask if the library installed with Intel 2022 is available on hera? Thanks.

**Comment by jkbk2004** [comment]
@junwang-noaa @Hang-Lei-NOAA I tried to install hpc-stack with intel22 on /work/noaa/epic-ps/jongkim/stacks/libs-intel2022.1.2. I may go ahead to do butterfly test on orion first. I am trying to install hpc-stack with intel2022 on Cheynne as well.

**Comment by Hang-Lei-NOAA** [comment]
@Jong Kim - NOAA Affiliate ***@***.***> @Jun Wang - NOAA Federal
***@***.***>  Library installed with Intel 2022 is available on
hera. For both gfsv16.2 and standard official installations. If extra
versions are needed, please let me know

**Comment by junwang-noaa** [comment]
@Hang-Lei-NOAA I got the following error when loading the libraries with intel 2022 on hera in ufs-weather-model develop branch:

module use /scratch2/NCEPDEV/nwprod/hpc-stack/libs/hpc-stack/modulefiles/stack
module load hpc/1.1.0
module load hpc-intel/2022.1.2
module load hpc-impi/2022.1.2

error:
Lmod has detected the following error: These module(s) or extension(s) exist
but cannot be loaded as requested: "jasper/2.0.22",
"mapl/2.11.0-esmf-8.2.1b04", "g2/3.4.2", "esmf/8.2.1b04", "pio/2.5.2",
"gftl-shared/v1.3.3", "crtm/2.3.0"

Would you please install those libraries with intel 2022? Those modules are currently used in ufs-weather-model develop branch, we hope to have a clean compiler update before changing any of the libraries. Thanks

**Comment by jkbk2004** [comment]
@junwang-noaa intel2022 hpc-stack on Cheyenne: /glade/scratch/jongkim/stacks/libs-intel2022.1.2/modulefiles/stack. I will run the butterfly test on both Orion and Cheyenne.

**Comment by Hang-Lei-NOAA** [comment]
@Jun Wang - NOAA Federal ***@***.***> the new installation with
intel/2022.1.2 was only installed with the latest hpc-stack module
versions. so, for jasper, please  module load jasper/2.0.25
all available libs are:
----------------------------------------------------------------------
/scratch2/NCEPDEV/nwprod/hpc-stack/libs/hpc-stack/modulefiles/mpi/intel/2022.1.2/impi/2022.1.2
-----------------------------------------------------------------------
   atlas/ecmwf-0.24.1    esmf/8_0_1        fckit/ecmwf-0.9.2    fms/2021.04
(D)    madis/4.3                mapl/2.7.3-esmf-8_1_1 (D)    nemsio/2.5.2
     nemsiogfs/2.5.3 (D)    pio/2.5.3    upp/10.0.9   (D)
   eckit/ecmwf-1.16.0    esmf/8_1_1 (D)    fms/2021.03          hdf5/1.10.6
(D)    mapl/2.7.3-esmf-8_0_1    ncio/1.0.0            (D)    nemsio/2.5.4
(D)    netcdf/4.7.4    (D)    upp/8.1.0    wrf_io/1.2.0
---------------------------------------------------------------------------
/scratch2/NCEPDEV/nwprod/hpc-stack/libs/hpc-stack/modulefiles/compiler/intel/2022.1.2
---------------------------------------------------------------------------
   bacio/2.4.1 (D)    crtm/2.4.0   (D)    g2tmpl/1.10.0      (D)
 hdf5/1.10.6              jasper/2.0.25     (D)    libpng/1.6.37
 netcdf/4.7.4           sp/2.3.3       (D)    w3nco/2.4.1    (D)
 zlib/1.2.11
   bufr/11.4.0        g2/3.4.5     (D)    gfsio/1.4.1        (D)
 hpc-impi/2022.1.2 (L)    jpeg/9.1.0               nccmp/1.8.9.0 (D)
 prod_util/1.2.2 (D)    szip/2.1.1     (D)    wgrib2/2.0.7   (D)
   bufr/11.5.0 (D)    g2c/1.6.4    (D)    gftl-shared/v1.3.0
 ip/3.3.3          (D)    landsfcutil/2.4.1 (D)    nco/5.0.6     (D)
 sfcio/1.4.1     (D)    udunits/2.2.28 (D)    wgrib2/2.0.8
   cdo/1.9.8   (D)    g2tmpl/1.9.1        grib_util/1.2.3    (D)
 ip2/1.1.2         (D)    libpng/1.6.35     (D)    nemsio/2.5.4
sigio/2.3.2     (D)    w3emc/2.9.2    (D)    yafyaml/v0.5.1
------------------------------------------------------------------------------------
/scratch2/NCEPDEV/nwprod/hpc-stack/libs/hpc-stack/modulefiles/core
-------------------------------------------------------------------------------------
   boost-headers/1.68.0        ecbuild/ecmwf-3.6.1          (D)
 esma_cmake/v3.4.3    hpc-intel/18.0.5.274           hpc-miniconda3/4.5.12
             json/3.9.1
   cmake/3.20.0         (D)    ecbuild/jcsda-release-stable
 gsl-lite/0.37.0      hpc-intel/2022.1.2    (L,D)    hpc-miniconda3/4.6.14
      (D)    pybind11/2.5.0
   cmakemodules/v1.2.0         eigen/3.3.7
hpc-gnu/9.2.0        hpc-intelpython/3.6.8
 json-schema-validator/2.1.0

**Comment by junwang-noaa** [comment]
@Hang-Lei-NOAA We need to evaluate the impact of intel compiler version change before we can use the modules in ufs-weather-model. I do see new versions of jasper, g2, pio and crtm, do you know if they will change results compared to previous versions? If yes, then we still need the previous versions. 

Also the gftl-shared/v1.3.3, esmf/8_2_1bs04 and MAPL/2.11.o are still missing.

**Comment by Hang-Lei-NOAA** [comment]
@Jun Wang - NOAA Federal ***@***.***> You can try this new versions
to see of the result are changed. e.g.GSI tested crtm 2.4.0. the results
are matching the 2.3.0.
But, per your request, I will add old versions of jasper, g2, pio and crtm,
and gftl-shared/v1.3.3, esmf/8_2_1bs04 and MAPL/2.11.0 today.

**Comment by Hang-Lei-NOAA** [comment]
@Jun Wang - NOAA Federal ***@***.***> all old versions and
previously mentioned extra versions, have been added on hera installation.
Please try it.

**Comment by jkbk2004** [comment]
@junwang-noaa @Hang-Lei-NOAA I am checking the difference of intel2021 and intel2022 on Cheyenne. The result seems to be identical. It is interesting. I will double check.  Orion que is very slow again.

**Comment by Hang-Lei-NOAA** [comment]
As the Hera manager Raghu mentioned, the intel/2022 addressed some bugs in
the intel/2021. Should be the same result.

**Comment by jkbk2004** [comment]
@Hang-Lei-NOAA Yeah, it seems like that's the case.

**Comment by junwang-noaa** [comment]
@Hang-Lei-NOAA @KyleNevins-NOAA  I tested Intel 2022.1.2 on hera it is working fine. Would you please install it on other platforms (dell, cray, jet, gaea)? Thank you!

**Comment by KyleNevins-NOAA** [comment]
> @Hang-Lei-NOAA @KyleNevins-NOAA I tested Intel 2022.1.2 on hera it is working fine. Would you please install it on other platforms (dell, cray, jet, gaea)? Thank you!

Hi @junwang-noaa, just wanted to check but I think you were looking for @kgerheiser ?

**Comment by junwang-noaa** [comment]
Sorry, Yes, this is for  @kgerheiser

**Comment by Hang-Lei-NOAA** [comment]
We will expand it on all platforms gradually. Space is still an issue for
some platforms, e.g. wcoss dell, since the removal of old was postponed.

**Comment by junwang-noaa** [comment]
@Hang-Lei-NOAA May I ask if the Intel 2022 is available on dell?

**Comment by Hang-Lei-NOAA** [comment]
@Jun Wang - NOAA Federal ***@***.***> wcoss will be off in a couple
months. We did not install it on dell.

**Comment by junwang-noaa** [comment]
@Hang-Lei-NOAA This means we will drop ufs-weather-model develop branch on dell any more when we move to Intel 2022. @arunchawla-NOAA FYI.

**Comment by arunchawla-NOAA** [comment]
Intel 2022 is not available on wcoss2, we will only have intel 19 there initially
 Does ufs-weather-model need intel 2022 in develop ?

**Comment by junwang-noaa** [comment]
Yes, there is a slowness issue with impi 18-19. We are trying to move to
Intel 2022 on all the platforms besides the intel 2018/2019 is not
supported on other R&D platforms.

**Comment by Hang-Lei-NOAA** [comment]
We may directly ask helpdesk to install compiler intel/2022.1 on wcoss2.

**Comment by junwang-noaa** [comment]
@arunchawla-NOAA Can we ask GRIT to install Intel/2022.1 on wcoss2?

**Comment by edwardhartnett** [comment]
@kgerheiser and @climbfuji did you have any trouble building HDF5 with the intel compilers? One user has reported problems with this on the netCDF site...

Also, what versions of HDF5 do you build? I assume 1.10.7 and 1.12.1?

**Comment by kgerheiser** [comment]
@edwardhartnett, no I don't think so. We build using Intel just about everywhere normally with NetCDF 4.7.4 and HDF5 1.10.6. And recently we've also tried NetCDF 4.8.1 and HDF5 1.12.1.

**Comment by climbfuji** [comment]
No problem at all building those versions on any of the platforms, with Intel versions 18 up to oneapi 2022.0.2 (note: we are still using the classic compilers, not `icx`/`ifx`).

**Comment by Hang-Lei-NOAA** [comment]
My understanding is that Edward recommend that UFS and GFS upgrade their
NETCDF &HDF5 versions to 1.10.7 and 1.12.1
@Jun Wang - NOAA Federal ***@***.***> please consider this

**Comment by arunchawla-NOAA** [comment]
GDIT has a huge backlog. I will make a help desk request but do not expect that to happen any time soon till the transition to operations happens formally

**Comment by junwang-noaa** [comment]
@Hang-Lei-NOAA The mpiall2all slowness in Intel 2018/2019 has nothing to do with Netcdf/HDF5, It was tested that the mpiall2all is fast in impi 2021/2022. We need impi 2021/2022 to continue the ESMF managed threading capability in UFS.

**Comment by junwang-noaa** [comment]
@kgerheiser May I ask if the hpc-stack with Intel 2022 is installed on jet and gaea? Thanks

**Comment by kgerheiser** [comment]
Hey @junwang-noaa I am working on that today.

**Comment by kgerheiser** [comment]
Just about done. A few more libraries to install because of an issue with the firewall on Jet.

**Comment by Hang-Lei-NOAA** [comment]
Have to get these blocked packages passed to jet through winscp or rsync,
then install.

**Comment by kgerheiser** [comment]
Jet and Orion have 2022.1.2, and Gaea with Intel 2021.3.0 has been completed.

**Comment by junwang-noaa** [comment]
@kgerheiser Thank you very much! I will run RT test on these platforms.

**Comment by jkbk2004** [comment]
@kgerheiser @junwang-noaa I will confirm on Cheyenne as well: module use /glade/work/epicufsrt/GMTB/tools/intel/2022.1/hpc-stack-v1.2.0_6eb6/modulefiles/stack

**Comment by junwang-noaa** [comment]
@kgerheiser with gnu compiler on hera, the g2/3.4.5 and pio/2.5.3 are not available,vwould you pelase install them? I got following error:

but cannot be loaded as requested: "g2/3.4.5", "pio/2.5.3" 

@jkbk2004 @MinsukJi-NOAA FYI.

**Comment by kgerheiser** [comment]
@Hang-Lei-NOAA Hera

**Comment by Hang-Lei-NOAA** [comment]
g2/3.4.5 and pio/2.5.3 have been added. Thanks!

**Comment by junwang-noaa** [comment]
@KyleNevins-NOAA @Hang-Lei-NOAA Would you please install hpc-stack with impi 2021.1.3 on hera? We had some issue with a 
 ufs-weather-model test using impi 2022.1.2, please see https://github.com/ufs-community/ufs-weather-model/pull/1228. Thank you very much for helping!

**Comment by Hang-Lei-NOAA** [comment]
Hi, @Jun Wang - NOAA Federal ***@***.***>

Could you please provide me of list of library versions that you required
in 2021.3 installation?
I have moved some of the old installation into my noscrub to release space
for this installation.

**Comment by junwang-noaa** [comment]
@Hang-Lei-NOAA Thanks. Here is the list:
esmf/8.3.0b09
esmf/8.3.0b09-debug
mapl/2.11.0-esmf-8.3.0b09
hdf5/1.10.6
netcdf/4.7.4
pio/2.5.3

**Comment by Hang-Lei-NOAA** [comment]
@Jun Wang - NOAA Federal ***@***.***>
The intel/2021.3.0 series has been installed on hera. Please test and let
me know if any changes need to be made.
Since the hera management said that 2022.1 is a more stable and
supported compiler. If intel/2021.3 installations cannot solve your issue.
I prefer to remove them to save space. Please let me know. Regards

**Comment by junwang-noaa** [comment]
@Hang-Lei-NOAA Thank you very much! The issue with impi 2022 was identified, we are trying to see if a workaround will resolve the issue. I will let you know if we need the intel/2021.3 soon. Thanks!

**Comment by Hang-Lei-NOAA** [comment]
@Jun Wang - NOAA Federal ***@***.***> I will try it. May we have to
remove some old stuff to release the space for a new set.
